pkg/segment/segment.go

```diff
@@ -0,0 +1,153 @@
+package segment
+
+import (
+	"errors"
+	"fmt"
+	"github.com/openshift/odo/pkg/preference"
+	"github.com/openshift/odo/pkg/version"
+	"github.com/pborman/uuid"
+	"golang.org/x/term"
+	"gopkg.in/segmentio/analytics-go.v3"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+// Writekey will be the API key used to send data to the correct source on Segment
+var WriteKey = "4xGV1HV7K2FtUWaoAozSBD7SNCBCJ65U"
+
+type Client struct {
+	segmentClient     analytics.Client
+	config            *preference.PreferenceInfo
+	telemetryFilePath string
+}
+
+func NewClient(config *preference.PreferenceInfo) (*Client, error) {
+	homeDir, _ := os.UserHomeDir()
+	return newCustomClient(config,
+		filepath.Join(homeDir, ".redhat", "anonymousId"),
+		analytics.DefaultEndpoint,
+	)
+}
+
+func newCustomClient(config *preference.PreferenceInfo, telemetryFilePath string, segmentEndpoint string) (*Client, error) {
+	client, err := analytics.NewWithConfig(WriteKey, analytics.Config{
+		Endpoint: segmentEndpoint,
+		DefaultContext: &analytics.Context{
+			IP: net.IPv4(0, 0, 0, 0),
```

Comment:
why aren't we getting the actually IP address here? or we actually shouldn't?

Reply:
Right, we shouldn't. We cannot collect any Personally Identifiable Information(PII).

Reply:
then why even set it?

Reply:
From what I read [here](https://segment.com/docs/connections/spec/common/#context), if we don't set the context with IP as we have done right now, it will take the current user's IP and we don't want that.
I will confirm this with PK and let you know.

Reply:
I confirmed this with PK and he agrees with the reason I have given here.